DetectDynamicJS.py

```diff
@@ -81,14 +81,17 @@ def doPassiveScan(self, baseRequestResponse):
             requestHeaders = request.tostring()[:requestBodyOffset].split('\r\n')
             requestBody = request.tostring()[requestBodyOffset:]
             modified_headers = "\n".join(header for header in requestHeaders if "Cookie" not in header)
-            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers+requestBody))
+            newResponse = self._callbacks.makeHttpRequest(baseRequestResponse.getHttpService(), self._helpers.stringToBytes(modified_headers + requestBody))
+            newResponseInfo = self._helpers.analyzeResponse(newResponse.getResponse())
+            if not (self.hasValidStatusCode(newResponseInfo.getStatusCode()) and self.hasBody(newResponseInfo.getHeaders())):
```

Comment:
Yes, that option for false positive exists. Despite the original efforts to detect dynamic JavaScript, I'd like the extension to also detect cases now, where JavaScript contains potentially interesting data based on ambient authority information. So static JavaScript, that is only reachable when the user is authenticated should be returned as well. Maybe it is worth reporting that in an extra finding to avoid confusion. `Detected Authentication-Based JavaScript`.